django/utils/module_loading.py

```diff
@@ -7,7 +7,11 @@
 
 def cached_import(module_path, class_name):
     modules = sys.modules
-    if module_path not in modules:
+    if module_path not in modules or (
+        # Module is not fully initialize.
+        getattr(modules[module_path], '__spec__', None) is not None and
+        getattr(modules[module_path].__spec__, '_initializing', False) is True
+    ):
         import_module(module_path)
     return getattr(modules[module_path], class_name)
```

Comment:
I'm not sure if we can decruft this a bit?
```diff
def cached_import(module_path, class_name):
    # Check whether module is loaded and fully initialized.
    if not (
        module := sys.modules.get(module_path) and
        spec := getattr(module, '__spec__', None) and
        getattr(spec, '_initializing', False) is False
    ):
        module = import_module(module_path)
    return getattr(module, class_name)
```

You'll have to double check this - just knocked it up quickly.

Reply:
> I'm not sure if we can decruft this a bit?

It raises a `SyntaxError`.